submariner/cable-driver-protocols-enhancement.md

```diff
@@ -0,0 +1,155 @@
+# Cable Driver Protocol Support Enhancements
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline additional protocols and enhancements for the Submariner
+Cable Driver. These include the additional support of IP-in-IP for unencrypted connections
+between clusters and the support of IPSec transport mode.
+
+## Proposal
+
+### IP-in-IP Encapsulation
+
+This enhancement proposes to support IP-in-IP for un-encrypted connection between clusters. A
+new IP tunnelling interface `ipip-tunnel` will be added to the active gateway node of each cluster
+and it will act as the Tunnel End Point. The traffic which is destined to other clusters will be
+forwarded via this interface to the appropriate clusters.
+
+> **_NOTE:_** A single IP Tunnelling interface will be added, and tunnels managed using
+`# ip route add encap ip <args>` in order to avoid a point to point connection per tunnel.
+
+#### Design Details
+
+* A new interface "ipip-tunnel" will be added.
+* The IP tunnelling interface will be one-to-many, that is a single interface will be used to
+connect to all the joined clusters.
+* For the "ipip-tunnel" IP, the prefix "242" will be used, followed by the rest of the blocks
+from the endpoint private IP address. For example, if the private IP address is 10.2.96.1 the
+"ipip-tunnel" IP would be 242.2.96.1.
+* The routes will be added to forward all the Service and Pod CIDR traffic to the respective
+remote Tunnel IPs.
+* Table 100 shall be used to add the routes, which is otherwise used for host network traffic
+routes. We do not require any extra rules to handle host network traffic for the IPTun driver.
```

Comment:
```suggestion
* Table 100 shall be used to add the routes for inter-cluster communication.
```

Table 100 is actually used by [VxLAN cable driver](https://github.com/submariner-io/submariner/blob/8c4145c64e1391048ffad9e7553ad47be0e26ff9/pkg/cable/vxlan/vxlan.go#L48) to program inter-cluster routes. We use Table 150 to support [hostNetworking use-cases](https://github.com/submariner-io/submariner/blob/8c4145c64e1391048ffad9e7553ad47be0e26ff9/pkg/routeagent_driver/constants/constants.go#L51) from the route-agent driver.

I think we can continue to use Table 100 even if its used by VxLAN cable driver.

Reply:
I will update it